packages/storage-plus/src/item.rs

```diff
@@ -17,9 +17,9 @@ pub struct Item<'a, T> {
 }
 
 impl<'a, T> Item<'a, T> {
-    pub const fn new(storage_key: &'a [u8]) -> Self {
+    pub const fn new(storage_key: &'a str) -> Self {
         Item {
-            storage_key,
+            storage_key: storage_key.as_bytes(),
```

Comment:
Great to keep the internals the same. We will have some binary data (length prefixed stuff), so easiest just to change the constructors for nicer public API